pkg/resources/network_policy.go

```diff
@@ -151,149 +181,168 @@ func ReadContextNetworkPolicy(ctx context.Context, d *schema.ResourceData, meta
 		}
 	}
 
-	if err = d.Set("name", networkPolicy.Name); err != nil {
+	if err = d.Set("name", sdk.NewAccountObjectIdentifier(networkPolicy.Name).Name()); err != nil {
```

Comment:
Why is this necessary?

Reply:
Because .Name is a string and by doing NewXIdentifier(...).Name() we ensure we'll get the same quoting (always without because our identifier constructors trim them). Ultimately, `.Name` will be changed into identifier and this will be `Set("name", networkPolicy.Name.Name())`